addons/account/models/account_journal.py

```diff
@@ -43,6 +43,16 @@ def _get_bank_statements_available_sources(self):
 
     def _default_alias_domain(self):
         return self.env["ir.config_parameter"].sudo().get_param("mail.catchall.domain")
+    
+    def _default_invoice_reference_model(self):
+        """Get the invoice reference model according to the company's country."""
+        country_code = self.env.company.country_id.code
+        country_code = country_code and country_code.lower()
+        if country_code:
+            for model in self._fields['invoice_reference_model'].get_values(self.env):
+                if model.startswith(country_code):
+                    return model
+        return 'odoo'
```

Comment:
```
@api.depends('country_code')
def _compute_invoice_reference_model(self):
    for journal in self:
        invoice_reference_model = False
        for model in self._fields['invoice_reference_model'].get_values(self.env):
            if model.startswith(country_code):
                invoice_reference_model = model
                break;
            if not invoice_reference_model:
                invoice_reference_model = 'odoo'
            journal.invoice_reference_model = invoice_reference_model

```

```
invoice_reference_model = fields.Selection(
    selection=[('odoo', 'Odoo'),('euro', 'European')],
    string='Communication Standard',
    store=True, readonly=False,
    compute='_compute_invoice_reference_model',
    help="You can choose different models for each type of reference. The default one is the Odoo reference.")
```

Reply:
The purpose of this is to backport https://github.com/odoo/odoo/commit/1493a01b5df1339765360a059a466ecfd2a10494.
@qdp-odoo should I make this change in master too ?

Reply:
@jbw-odoo I didn't understand it was a backport. I let @qdp-odoo to decide ;)